synapse/server.py

```
@@ -303,10 +301,20 @@ async def get_status():
     os.environ["SYNAPSE_PORT"] = str(args.port)
 
-    uvicorn.run(app, host="0.0.0.0", port=args.port)
+    # Configure SSL if certificates provided
+    ssl_config = {}
+    if args.ssl_cert and args.ssl_key:
+        ssl_config["ssl_certfile"] = args.ssl_cert
+        ssl_config["ssl_keyfile"] = args.ssl_key
+        os.environ["SYNAPSE_USE_HTTPS"] = "true"
+        print(f"HTTPS enabled with certificate: {args.ssl_cert}")
+
+    uvicorn.run(app, host="0.0.0.0", port=args.port, **ssl_config)
```

Ruff (0.14.10)

Possible binding to all interfaces

(S104)